stdlib/Dates/src/deprecated.jl

```diff
@@ -66,5 +66,10 @@ for op in (:+, :-)
     end
 end
 
-@deprecate argerror(msg::String) ArgumentError(msg) false
-@deprecate argerror() nothing false
+function Base.round(x::TimeTypeOrPeriod, ::Type{P}, r::RoundingMode=RoundNearestTiesUp) where P <: Period
```

Comment:
These methods need to actually be deprecated.

Reply:
Can I get some help in deprecation?

Reply:
There is nothing special about the file deprecated.jl; what makes the methods deprecated is the use of the `@deprecate` macro. You can look at the docstring `?@deprecate` for more info. To define a deprecated method for a function defined in another module you need to import that function:
```julia
import Base: round
@deprecate round(...) round(...)
```
To make sure the deprecations are working, start Julia with `--depwarn=yes` and make sure that deprecations are emitted when expected
```julia
julia> round(today(), Week)
┌ Warning: `round(x, T::Type{<:Period})` is deprecated, use `round(T, x)` instead.
[...]
```

Reply:
@LilithHafner Thanks for this. I was trying to figure this out.